src/components/tl/cuda/tl_cuda.h

```diff
@@ -113,19 +114,23 @@ typedef struct ucc_tl_cuda_sync_data {
 } ucc_tl_cuda_sync_data_t;
 
 typedef struct ucc_tl_cuda_mem_info {
-    void               *ptr;
-    size_t              length;
-    size_t              offset;
-    cudaIpcMemHandle_t  handle;
+    void              *ptr;
+    size_t             length;
+    size_t             offset;
+    cudaIpcMemHandle_t handle;
 } ucc_tl_cuda_mem_info_t;
 
 typedef struct ucc_tl_cuda_sync {
-    uint32_t                 seq_num[2];
-    ucc_tl_cuda_mem_info_t   mem_info_src;
-    ucc_tl_cuda_mem_info_t   mem_info_dst;
-    cudaEvent_t              ipc_event_local;
-    cudaIpcEventHandle_t     ev_handle;
-    ucc_tl_cuda_sync_data_t  data[1];
+    uint32_t                seq_num[2];
+    ucc_tl_cuda_mem_info_t  mem_info_src;
+    ucc_tl_cuda_mem_info_t  mem_info_dst;
+    cudaEvent_t             ipc_event_local;
+    cudaIpcEventHandle_t    ev_handle;
+    ucc_tl_cuda_sync_data_t data[UCC_TL_CUDA_MAX_PEERS];
```

Comment:
@Sergei-Lebedev - I have changed this from `data[1];` to `data[UCC_TL_CUDA_MAX_PEERS];` to avoid index-out-of-range error when running alltoallv (when perfoming cudaStreamWaitEvent). I'm surprised it works fine for alltoall, please let me know if this change looks okay to you.

Reply:
this is fixed by moving around the data structure of `ucc_tl_cuda_sync_t` as this PR adds new fields like `src_cnts`.